src/Services/DocumentTemplates/DocumentTemplateService.php

```diff
@@ -12,12 +12,331 @@
+        );
+        while ($row = sqlFetchArray($query_result)) {
+            $groups = explode('|', $row['patient_groups']);
+            foreach ($groups as $group) {
+                $results[$group][] = $row;
+            }
+        }
+        return $results;
+    }
+
+    /**
+     * @param $group_array
+     * @return array
+     */
+    public function getPatientsByGroup($group_array): array
+    {
+        $results = [];
+        foreach ($group_array as $group) {
+            $search = '%' . $group . '%';
+            $query_result = sqlStatement(
+                'SELECT pid, pubpid, fname, mname, lname, DOB, patient_groups FROM patient_data WHERE patient_groups <> "" AND patient_groups LIKE ? ORDER BY `lname`',
+                array($search)
+            );
+            while ($row = sqlFetchArray($query_result)) {
+                if (is_array($row)) {
+                    $results[$group][] = $row;
+                }
+            }
+        }
+        return $results;
+    }
+
+    /**
+     * @param $profile
+     * @return array
+     */
+    public function getPatientGroupsByProfile($profile): array
+    {
+        $rtn = sqlStatement('SELECT `profile`, `member_of`, `active` FROM `document_template_profiles` WHERE `profile` = ? AND `member_of` > ""', array($profile));
+        $profile_list = array();
+        while ($row = sqlFetchArray($rtn)) {
+            $profile_list[$row['profile']][] = $row;
+        }
+        return $profile_list;
+    }
+
+    /**
+     * @param $profile
+     * @return array
+     */
+    public function getTemplateListByProfile($profile): array
+    {
+        $rtn = sqlStatement('SELECT `template_id`, `category` FROM `document_template_profiles` WHERE `profile` = ? AND `template_id` > 0', array($profile));
+        $profile_list = array();
+        while ($row = sqlFetchArray($rtn)) {
+            $profile_list[$row['category']][] = $this->fetchTemplate($row['template_id']);
+        }
+        return $profile_list;
+    }
+
+    /**
+     * @param      $id
+     * @param null $template_name
+     * @return array|false|null
+     */
+    public function fetchTemplate($id, $template_name = null)
+    {
+        $return = null;
+        if (!empty($id)) {
+            $return = sqlQuery('SELECT * FROM `document_templates` WHERE `id` = ?', array($id));
+        } elseif (!empty($template_name)) {
+            $return = sqlQuery('SELECT * FROM `document_templates` WHERE `template_name` = ?', array($template_name));
+        }
+        return $return;
+    }
+
+    /**
+     * @param $profile_groups
+     * @return bool
+     */
+    public function savePatientGroupsByProfile($profile_groups): bool
+    {
+        sqlStatementNoLog('SET autocommit=0');
+        sqlStatementNoLog('START TRANSACTION');
+
+        try {
+            sqlQuery('DELETE From `document_template_profiles` WHERE `template_id` = 0');
```

Comment:
seems odd to do this with sqlQuery (sqlStatement perhaps instead) since that goal of that function is to return one row of data as an array. it will work, though.

Reply:
it's a Delete that returns int total or 0 so appropriate